syntax = "proto3";

package analysis.dvvp.proto;

message ProfAclConfig {
    string switch = 1;
    string output = 2;
    string device_id = 3; // This Filed is invalidated now.
    string aic_metrics = 4;
    string aicpu = 5;
    string l2 = 6;
    string storage_limit = 7;
    string hccl = 8;
    string msproftx = 9;
    string biu = 10;
    uint32 biu_freq = 11;
    string task_time = 12;
    string aiv_metrics = 13;
    string ascendcl = 14;
    string runtime_api = 15;
    uint32 sys_usage_freq = 16;
    uint32 sys_pid_usage_freq = 17;
    uint32 sys_cpu_freq = 18;
    uint32 sys_hardware_mem_freq = 19;
    string llc_mode = 20;
    uint32 sys_io_freq = 21;
    uint32 sys_interconnection_freq = 22;
    uint32 dvpp_freq = 23;
    string host_sys_usage = 24;
}

message ProfGeOptionsConfig {
    string output = 1;
    string training_trace = 2;
    string task_trace = 3;
    string aicpu = 4;
    string fp_point =5;
    string bp_point = 6;
    string aic_metrics = 7;
    string stars_acsq_task = 8;
    string stars_sub_task = 9;
    string ffts_thread_task = 10;
    string l2 = 11;
    string storage_limit = 12;
    string ffts_block = 13;
    string low_power = 14;
    string acc_pmu_mode = 15;
    string hccl = 16;
    string msproftx = 17;
    string biu = 18;
    uint32 biu_freq = 19;
    string task_time = 20;
    string aiv_metrics = 21;
    string runtime_api = 22;
    uint32 sys_usage_freq = 23;
    uint32 sys_pid_usage_freq = 24;
    uint32 sys_cpu_freq = 25;
    uint32 sys_hardware_mem_freq = 26;
    string llc_mode = 27;
    uint32 sys_io_freq = 28;
    uint32 sys_interconnection_freq = 29;
    uint32 dvpp_freq = 30;
    string host_sys_usage = 31;
}

message ProfAclInitConfig {
    string path = 1;
}
